app/controllers/dashboard/form/work_version_type_controller.rb

```diff
@@ -23,8 +23,15 @@ def create
 
       def update
         @resource = WorkVersion.includes(:work).find(params[:id])
+        original_work_type = @resource.work_type
         authorize(@resource)
         @resource.attributes = work_version_params
+        if original_work_type != @resource.work_type
+          fields_to_reset = WorkDepositPathway.new(@resource).fields_to_reset(original_work_type)
+          fields_to_reset.each do |field|
+            @resource.send(:"#{field}=", nil)
+          end
+        end
```

Comment:
I'm wondering if we could pull this logic out and put it in the WorkVersion class?

Reply:
This mostly worked, just had to change how the actual field was set to nil